pipeline/scripts/annotate_variants.py

```diff
@@ -0,0 +1,387 @@
+                else:
+                    try:
+                        depth = sample_info["AD"][1] / sample_info["AF"]  # Guess the depth from alt count and VAF
+                    except ZeroDivisionError:
+                        if sample_info["AD"][1] == 0:
+                            depth = 0
+                        else:
+                            depth = np.nan
+                variants_df.loc[idx, f'{label}_{kind}_depth'] = depth
+
+                if "FA" in sample_info:
+                    variants_df.loc[idx, f'{label}_{kind}_vaf'] = sample_info["FA"]
+                elif "AF" in sample_info:
+                    variants_df.loc[idx, f'{label}_{kind}_vaf'] = sample_info["AF"]
+
+                if "AD" in sample_info:
+                    ref_count, alt_count = sample_info["AD"]
+                else:
+                    ref_count = alt_count = np.nan
+
+                variants_df.loc[idx, f'{label}_{kind}_ref_count'] = ref_count
+                variants_df.loc[idx, f'{label}_{kind}_alt_count'] = alt_count
+
+                variants_df.loc[idx, f"{label}"] = True
+
+    for kind in ["normal", "tumor"]:
+        for suffix in ["ref_count", "alt_count", "depth", "vaf"]:
+            col = f'{label}_{kind}_{suffix}'
+            if variants_df[col].isnull().all():
+                print("Dropping col (all nan)", col)
+                del variants_df[col]
+
+    print(f"Annotated {label} variants: {variants_df[label].sum()} of {len(variants_df)}")
+    return variants_df
+
+
+def annotate_from_bam(bam_file, variants_df, label, min_mapq=10):
+    """
+    Function to count reads supporting reference and alternate alleles for given variants in a BAM file.
+
+    Parameters:
+    bam_file (str): Path to the BAM file.
+    variants_df (pd.DataFrame): DataFrame containing variants with columns 'contig', 'start', 'ref', 'alt'.
+    label (str): Label for the BAM file (used to name the count columns).
+
+    Returns:
+    pd.DataFrame: DataFrame with additional columns for read counts and depth.
+    """
+    # Open the BAM file
+    bam = pysam.AlignmentFile(bam_file, "rb")
+
+    # index the bam if needed
+    if not bam.has_index():
+        print(f"Indexing {bam_file}")
+        pysam.index(bam_file)
+        bam = pysam.AlignmentFile(bam_file, "rb")
+
+    # Initialize new columns in the DataFrame
+    variants_df[f'{label}_ref_count'] = 0
+    variants_df[f'{label}_alt_count'] = 0
+    variants_df[f'{label}_depth'] = 0
+    variants_df[f'{label}_vaf'] = np.nan
+
+    # Iterate over each variant
+    for idx, row in tqdm(variants_df.iterrows(), total=variants_df.shape[0],
+            desc=f"Annotating {label}"):
+        contig = row['contig']
+        start = row['start']
+        ref = row['ref']
+        alt = row['alt']
+
+        ref_count = 0
+        alt_count = 0
+        total_depth = 0
+
+        # Deal with varcode name mangling
+        possible_contigs = [contig, "chr" + contig]
+        if contig == "MT":
```

Comment:
I don't know actually. The varcode name mangling stuff is pretty unfortunate. This is just a big hack to figure out chr names that correspond to the BAM files. If there's a clearner way that is simple I'd love to do that. In the longer run it would probably be better to have vaxrank output files with contig names that correspond to the BAM reference names.